spec/services/metrics/weekly/review_turaround/per_user_project_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+RSpec.describe Metrics::Weekly::ReviewTurnaround::PerUserProject do
+  describe '.call' do
+    let(:project) { create(:project) }
+    let(:users_project) { create(:users_project, project: project) }
+
+    context 'generating a weekly metric for a user who has made his first review' do
+      let!(:metric) { create(:metric, ownable: users_project, value_timestamp: Time.zone.today) }
+      let(:weekly_metric_just_created) { Metric.find_by(interval: :weekly) }
+      it 'creates the metric with the same value the current daily metric' do
+        described_class.call
+        expect(metric.value).to eq(weekly_metric_just_created.value)
+      end
+
+      it 'creates just one weekly metric per daily metric' do
+        expect { described_class.call }.to change { Metric.count }.by(1)
+      end
+    end
+
+    context 'generating a weekly metric for a user who has previous reviews' do
+      let!(:current_time) { Time.zone.parse('2020-04-29') }
+
+      let!(:weekly_metric) do
+        create(:metric,
+               ownable: users_project,
+               value_timestamp: (current_time - 1.day),
+               value: 120,
+               interval: :weekly)
+      end
+
+      let!(:daily_metric) do
+        create(:metric,
+               ownable: users_project,
+               value: 60,
+               value_timestamp: current_time)
+      end
+
+      let(:current_number_day) { current_time.wday }
+
+      let!(:avg) { (weekly_metric.value + daily_metric.value) / current_number_day }
+
+      before do
+        allow(Time).to receive(:now) { current_time }
```

Comment:
you can freeze_time instead of mocking it https://api.rubyonrails.org/v5.2.4.2/classes/ActiveSupport/Testing/TimeHelpers.html#method-i-freeze_time

Reply:
but it is possible to set a specific date? because i need that this date don't be weekend

Reply:
i don't know if i should use freeze_time and travel in that case